### ISSUE TITLE
provisioner/chef: Error parsing vault_json: json: cannot unmarshal array into Go value of type string

### ISSUE THREAD

**Issue by jmccann** [hashicorp/terraform#9105]
### Terraform Version
Terraform v0.7.4

### Affected Resource(s)
- chef

### Debug Output
```
Errors:

  * openstack_compute_instance_v2.test: Error parsing vault_json: json: cannot unmarshal array into Go value of type string
Error!
exit status 1
[info] build failed (exit code 1)
```

### Expected Behavior
It should have worked with no errors.

### Actual Behavior
It returned the error above

### Steps to Reproduce
Run terraform with chef provisioner that has `vault_json` that contains an array.

Example:
```
  provisioner "chef" {
    attributes_json = <<-EOF
    {
      "rsyslog": {
        "server_ip": "192.168.1.23"
      }
    }
    EOF
    environment = "test"
    node_name = "test-server"
    recreate_client = true
    run_list = ["role[test]"]
    server_url = "https://test.company.com/organizations/test"
    ssl_verify_mode = "verify_none"
    user_name = "test-validator"
    user_key = "${base64decode(var.test_key)}"
    vault_json = <<-EOF
    {
      "test": [
        "test"
      ]
    }
    EOF
  }
```

### Important Factoids
`vault_json` was recently added to `chef` provisioner.

I think the bad line of code causing this issue is @ https://github.com/hashicorp/terraform/blob/master/builtin/provisioners/chef/resource_provisioner.go#L355 ... specifically the `string` at the end of the line should be `interface{}` like it is for `attributes_json` @ https://github.com/hashicorp/terraform/blob/master/builtin/provisioners/chef/resource_provisioner.go#L347

### References
`vault_json` argument added as part of #8577 

@svanharmelen What are your thoughts on this?

**Comment by svanharmelen** [comment]
@jmccann yes, I didn't know the value could be both a string and an array of strings. Will indeed need to update that line, but a few others as well. Will put in a fix later today... Thx for the report!

**Comment by svanharmelen** [comment]
@jmccann the fix is in PR #9114 Again it would be great if you can give it a test run to verify my work! If you are able to do so, then please also check if the user key is deleted before the Chef run is started... Thx!
